src/config.c

```diff
@@ -862,6 +934,50 @@ void RSConfigExternalTrigger_Register(RSConfigExternalTrigger trigger, const cha
   RSGlobalConfigTriggers[numTriggers++] = trigger;
 }
 
+#ifdef MT_BUILD
+
+void UpgradeDeprecatedMTConfigs() {
+  RSConfigVar *workers = findConfigVar(&RSGlobalConfigOptions, "WORKERS");
+  RSConfigVar *minOperationWorkers = findConfigVar(&RSGlobalConfigOptions, "MIN_OPERATION_WORKERS");
+  if (workers->flags & RSCONFIGVAR_F_MODIFIED || minOperationWorkers->flags & RSCONFIGVAR_F_MODIFIED) {
+    return; // New configurations were set. Ignore the deprecated ones.
+  }
+  RSConfigVar *mtMode = findConfigVar(&RSGlobalConfigOptions, "MT_MODE");
+  RSConfigVar *workerThreads = findConfigVar(&RSGlobalConfigOptions, "WORKER_THREADS");
+  if (!(mtMode->flags & RSCONFIGVAR_F_MODIFIED) && !(workerThreads->flags & RSCONFIGVAR_F_MODIFIED)) {
+    return; // No deprecated configurations were set.
+  }
+
+  // We now know that deprecated configurations were set, and new configurations were not set.
+  if ((mt_mode_config == MT_MODE_OFF && numWorkerThreads_config != 0) ||
+      (mt_mode_config != MT_MODE_OFF && numWorkerThreads_config == 0)) {
+    RedisModule_Log(RSDummyContext, "warning",
+                    "Inconsistent configuration: MT_MODE `%s` and WORKER_THREADS `%lu`. Ignoring "
+                    "the deprecated configurations.",
+                    MTMode_ToString(mt_mode_config), numWorkerThreads_config);
+    return; // Inconsistent configuration. Ignore the deprecated configurations.
+  }
+
+  // Set the new configurations based on the deprecated ones.
+  switch (mt_mode_config) {
+    case MT_MODE_OFF:
+      RedisModule_Log(RSDummyContext, "warning",
+                      "Setting `MIN_OPERATION_WORKERS` to 0 due to explicit `MT_MODE_OFF`, "
+                      "overriding the default of " STRINGIFY(MIN_OPERATION_WORKERS));
+      RSGlobalConfig.numWorkerThreads = 0;
+      RSGlobalConfig.minOperationWorkers = 0;
+      break;
+    case MT_MODE_FULL:
+      RSGlobalConfig.numWorkerThreads = numWorkerThreads_config;
```

Comment:
What is the assumption on the default `minOperationThreads` here? To have the exact 2.8 behavior, we should set it to `numWorkerThreads_config` as well no?

Reply:
I think the “other” configuration should keep its default value, regardless of what it is. I’m more worried about losing the new default behavior on RCP after an upgrade from the preview to GA, than enabling the operation thread. This is both the default behavior when using 2.10 and we didn’t have the option to combine the “full” and “operations” modes before

Reply:
Emphasis that operation threads are set by the new defaults anyway regardless of the old configs if we are on MT_MODE full